src/wallet/rpcwallet.cpp

```diff
@@ -1348,8 +1348,8 @@ UniValue sendmany(const UniValue& params, bool fHelp)
 
     // Send
     CReserveKey keyChange(pwalletMain);
-    CAmount nFeeRequired = 0;
-    int nChangePosRet = -1;
+    CAmount nFeeRequired = -1; // never used
```

Comment:
It makes me nervous to assign negative values to `CAmount` variables. Since this was already being set to zero, it wasn't uninitialized; why is this change needed?

Reply:
The particular change on this line isn't needed, because as the comment says, the value is never used. However, the change on 314 is needed and in that case, the initialization value can't be in-band: it has to be distinguishable from any valid `CAmount` value. I could have changed all of these uses to `std::optional<CAmount>` (which arguably is more semantically correct), but that would have been a more disruptive change than I wanted to make. I was being trying to be very systematic and ensure that exactly the same pattern is used for every call to `CreateTransaction`, which made it easier to have confidence that all cases had been caught.

Reply:
Perhaps a safer way to handle these "never used" cases would be to declare them within a narrower scope, so that it's easier to see that they're not used outside that scope.

Reply:
I followed this suggestion; please re-review.